index.html

```diff
@@ -854,23 +862,49 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 '!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
 entry buffer</a>.</li>
 <li>Increase <a>resource timing buffer current size</a> by 1.</li>
+<li>Return.</li>
 </ol>
 </li>
-<li>If <a>resource timing buffer current size</a> is greater than
-or equal to <a>resource timing buffer size limit</a> and the
-<a>resource timing buffer full flag</a> is false, run the following
+<li>If <a>resource timing buffer full flag</a> is false, run the following
 substeps:
-<ol style='list-style-type: lower-latin;' data-link-for=
-"Performance">
-<li>Set the <a>resource timing buffer full flag</a> to true.</li>
+<ol style="list-style-type: lower-latin;">
+<li>Set <a>resource timing buffer full flag</a> to true.</li>
+<li>Queue a task to run <a>fire a buffer full event</a>.</li>
+</ol>
+<li>If <a>resource timing secondary buffer current size</a> is less than
+<a>resource timing buffer size limit</a>, add <i>new entry</i> to the
+<a>secondary entry buffer</a> and increment <a>resource timing secondary buffer
+current size</a> by 1.</li>
+</ol>
+
+<p>To <dfn>fire a buffer full event</dfn>, run the following steps:</p>
+<ol>
 <li><a data-cite="!DOM/#concept-event-fire">Fire an event</a>
 named <code>resourcetimingbufferfull</code> at the <a data-cite=
 "!HR-TIME-2/#idl-def-performance">Performance</a> object, with its
 <code>bubbles</code> attribute initialized to true.</li>
+<li>While <a>resource timing buffer current size</a> is smaller than
+<a>resource timing buffer size limit</a> and <a>secondary entry buffer</a> is
+not empty, run the following substeps:
+<ol>
+<li>Let <i>entry</i> be the oldest <a>PerformanceResourceTiming</a> in
+<a>secondary entry buffer</a>.</li>
+<li>Add <i>entry</i> to <a
+data-cite='!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
+entry buffer</a>.</li>
+<li>Increment <a>resource timing buffer current size</a> by 1.</li>
+<li>Remove <i>entry</i> from <a>secondary entry buffer</a>.</li>
```

Comment:
Looks like this is missing a step to fire `resourcetimingbufferfull` once the primary buffer becomes full again?

Reply:
because the secondary buffer is bounded, and we assume developers do the right thing (either clear or double the buffer size), there's no need to fire the event again from here.

Reply:
Note that one behavior change with this PR is that the event doesn't fire when the buffer is full, but fired when it *overflows*.

Reply:
Okay, that needs to be changed once the secondary buffer's limit is lifted.

Reply:
Even if we intend to keep an unbounded secondary buffer, for consistency we might want to fire the additional events async. I see no reason why this one would need be sync. So essentially we would just call the "add a PerformanceResourceTiming entry" algorithm.

Reply:
Making those secondary events async would be fine with me.